### PR TITLE
Add EU868 specific datarates for PoC packets

### PR DESCRIPTION
**Pull request by amirhaleem**
This PR adds on to #521 by adding additional EU868 specific datarates for PoC, based on the guidance from Semtech here: https://lora-developers.semtech.com/library/tech-papers-and-guides/the-book/packet-size-considerations/

### CODE REVIEW

File: src/poc/miner_onion_server.erl [hunk 1]
```diff
@@ -345,7 +345,8 @@ decrypt(Type, IV, OnionCompactKey, Tag, CipherText, RSSI, SNR, Frequency, Channe
                 true ->
                     %% the fun below will be executed by miner_lora:send and supplied with the localised lists of channels
                     ChannelSelectorFun = fun(FreqList) -> lists:nth((IntData rem 8) + 1, FreqList) end,
-                    Spreading = spreading(erlang:byte_size(Packet)),
+                    {_, Region} = miner_lora:region(),
```
Comment (Vagabond) [inline src/poc/miner_onion_server.erl:348]:
```suggestion
                    {ok, Region} = miner_lora:region(),
```

**Review by Vagabond — APPROVED** [review]

File: src/poc/miner_onion_server.erl [hunk 2]
```diff
@@ -392,15 +393,20 @@ try_decrypt(IV, OnionCompactKey, OnionKeyHash, Tag, CipherText, ECDHFun, Chain)
             {error, too_many_pocs}
     end.
 
--spec spreading(integer()) -> string().
-spreading(Len) when Len < 12 ->
+-spec spreading(integer(), atom()) -> string().
+spreading(Len, _) when Len < 12 ->
     "SF10BW125";
-spreading(Len) when Len < 54 ->
+spreading(Len, 'EU868') when Len < 52 ->
```
Comment (abhay) [inline src/poc/miner_onion_server.erl:399]: this atom should probably be a constant?

File: src/poc/miner_onion_server.erl [hunk 3]
```diff
@@ -392,15 +393,20 @@ try_decrypt(IV, OnionCompactKey, OnionKeyHash, Tag, CipherText, ECDHFun, Chain)
...
+    "SF9BW125";
+spreading(Len, _) when Len < 126 ->
+    "SF8BW125";
+spreading(Len, 'EU868') when Len < 223 ->
     "SF8BW125";
-spreading(Len) when Len < 243 ->
+spreading(Len, _) when Len < 243 ->
     "SF7BW125";
-spreading(_) ->
-    %% onion packets won't be this big, but this will top out around 180 bytes
```
Comment (abhay) [inline src/poc/miner_onion_server.erl:405]: keep this comment?

File: src/poc/miner_onion_server.erl [hunk 4]
```diff
@@ -392,15 +393,20 @@ try_decrypt(IV, OnionCompactKey, OnionKeyHash, Tag, CipherText, ECDHFun, Chain)
...
     "SF9BW125";
-spreading(Len) when Len < 126 ->
+spreading(Len, 'EU868') when Len < 116 ->
+    "SF9BW125";
+spreading(Len, _) when Len < 126 ->
+    "SF8BW125";
+spreading(Len, 'EU868') when Len < 223 ->
     "SF8BW125";
-spreading(Len) when Len < 243 ->
+spreading(Len, _) when Len < 243 ->
```
Comment (vihu) [inline src/poc/miner_onion_server.erl:409]: This clause is not required I think

**Review by vihu — APPROVED** [review]

**Pull request merged**